src/runtime_src/core/common/api/xrt_error.cpp

```diff
@@ -205,6 +206,27 @@ class error_impl
 public:
   error_impl(const xrt_core::device* device, xrtErrorClass ecl)
   {
+    //Code for new format; Binary array of error structs in sysfs
+    try {
+      std::vector<char> buf = xrt_core::device_query<xrt_core::query::xocl_errors>(device);
```

Comment:
Please use auto, the type is explicit in xrt_core::query::xocl_errors.

Reply:
updated